### DISCUSSION TITLE
Application Testnets

### DISCUSSION THREAD

**Discussion by fernandofcampos** [cosmos/community#91] [Chain Builders (Cosmos SDK)]
I've been trying to take the state from our mainnet and run tests against the state, following this guide: https://docs.cosmos.network/v0.50/build/building-apps/app-testnet

Unfortunately the guide seems to be incomplete, and I've been debugging and solving issues but wasn't able to make it work yet.. Does anyone have a working example to share?

**Comment by julienrbrt** [comment]
Hey! Where are you facing an issue?

If your issue is how to wire it properly in your chain, Ignite CLI just happened to have integrated this feature.
If you scaffold a chain there, you'll see how the wiring should be done in a real chain.

```sh
git clone https://github.com/ignite/cli
make install
ignite s chain gm
ignite s chain serve # wait a few blocks and exit
# change validator name in ignite `config.yaml`
ignite testnet in-place # see validators have changed
```
